### PR TITLE
Add get events by parameters endpoint

### PR DESCRIPTION
**Pull request by toby-knott-cko**
Adding the specs for the get events by query parameters endpoint

### CODE REVIEW

File: abc_spec/paths/events.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      schema:
+        type: string
+        pattern: "^(pay)_(\\w{26})$"
+        description: The identifier of a payment
+        example: pay_ok2ynq6ubn3ufmo6jsdfmdvy5q
+    - in: query
+      name: charge_id
+      schema:
+        type: string
+        pattern: "^(charge)_(\\w{26})$"
```
Comment (roman-pavlov-cko) [inline abc_spec/paths/events.yaml:22]: Pattern for charge Id is: "^(charge|charge_test)_(\\w{20})$"

File: abc_spec/paths/events.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        description: The identifier of a charge
+        example: charge_FC1919EE547L23CC6BE1
+    - in: query
+      name: track_id
+      schema:
+        type: string
+        pattern: "^(TRK)_(\\w{26})$"
```
Comment (roman-pavlov-cko) [inline abc_spec/paths/events.yaml:29]: Please remove the pattern for Track Id. It a free format

File: abc_spec/components/schemas/Events/Payment_Id.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+type: object
```
Comment (roman-pavlov-cko) [inline abc_spec/components/schemas/Events/Payment_Id.yaml:1]: Just a suggestion: you could have defined the GetEventsResponse and reuse it for Charge Id and Payment Id.
The only diff is the example of event_type:
```
type:
          type: string
          description: The type of event
          example: "payment_approved"
```

**Review by roman-pavlov-cko — COMMENTED** [review]
Looks ok. Left some comments

File: abc_spec/components/schemas/Events/ChargeIdInvalidResponse.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+type: object 
+properties:
+  error_codes:
+    type: array
+    description: Array containing error codes
+    items:
+      type: string
+      example: "charge_id_invalid"
```
Comment (nikolay-tinchev-cko) [inline abc_spec/components/schemas/Events/ChargeIdInvalidResponse.yaml:8]: Leaving for visibility:
```{
    "request_id": "0HMA0CMCKVMI1:00000001",
    "error_type": "request_invalid",
    "error_codes": [
        "charge_id_invalid"
    ]
}```

File: abc_spec/components/schemas/Events/PaymentIdInvalidResponse.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+type: object 
+properties:
+  error_codes:
+    type: array
+    description: Array containing error codes
+    items:
+      type: string
+      example: "payment_id_invalid"
```
Comment (nikolay-tinchev-cko) [inline abc_spec/components/schemas/Events/PaymentIdInvalidResponse.yaml:8]: Leaving for visibility:
```{
    "request_id": "0HMA0CMCKVMI1:00000001",
    "error_type": "request_invalid",
    "error_codes": [
        "payment_id_invalid"
    ]
}```

**Comment by nikolay-tinchev-cko** [comment]
One more query parameter that has been omitted before should be added:

`/events?reference=ORD-5023-4E89`
The reference of a payment

File: abc_spec/components/schemas/Events/Charge_Id.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        _links:
+          type: object
+          properties:
+            self:
+              type: object
+              properties:
+                href:
+                  type: string
+                  description: The URL
+                  example: "https://qaapi.ckotech.co/broadcast/events/evt_a4aqkopnq65udagtnlpqsqr7uy"
```
Comment (nikolay-tinchev-cko) [inline abc_spec/components/schemas/Events/Charge_Id.yaml:49]: One final thing is that the _links urls have to be updated with the prod endpoint:
"_links": {
"self": {
"href": "https://api.checkout.com/events/evt_3nup2pts3emebenhtw6ky4frim"
},
"webhooks-retry": {
"href": "https://api.checkout.com/events/evt_3nup2pts3emebenhtw6ky4frim/webhooks/retry"
}
}

File: abc_spec/components/schemas/Events/Charge_Id.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            webhooks-retry:
+              type: object
+              properties:
+                href:
+                  type: string
+                  description: The URL
+                  example: "https://qaapi.ckotech.co/broadcast/events/evt_a4aqkopnq65udagtnlpqsqr7uy/webhooks/retry"
```
Comment (nikolay-tinchev-cko) [inline abc_spec/components/schemas/Events/Charge_Id.yaml:56]: One final thing is that the _links urls have to be updated with the prod endpoint:

"self": {
"href": "https://api.checkout.com/events/evt_3nup2pts3emebenhtw6ky4frim"
},
"webhooks-retry": {
"href": "https://api.checkout.com/events/evt_3nup2pts3emebenhtw6ky4frim/webhooks/retry"
}
}```

File: abc_spec/components/schemas/Events/Payment_Id.yaml (hunk 3 above)
Comment (toby-knott-cko) [reply abc_spec/components/schemas/Events/Payment_Id.yaml:1]: I tried to do that initially and just specify the examples separately, but when I did that the arrays appeared as objects in the example. I had the same issue with the 422 response samples. It's possible I've missed some obvious solution though.

**Review by nikolay-tinchev-cko — APPROVED** [review]
Looks good to me.

**Review by jack-hutchinson-cko — APPROVED** [review]

**Pull request merged**